tools/checklocks/analysis.go

```diff
@@ -843,11 +847,47 @@ func (pc *passContext) checkFunction(call callCommon, fn *ssa.Function, lff *loc
 	}
 }
 
+// isMutexType checks if a given type is sync.Mutex or sync.RWMutex.
+// Simplified implementation focusing on direct Named type check.
+func (pc *passContext) isMutexType(typ types.Type) bool {
+	// Fast-path: unwrap aliases and pointers.
+	seen := make(map[types.Type]struct{})
+	cur := typ
+	for {
+		if _, dup := seen[cur]; dup {
+			// Cycle guard, should never happen.
+			break
+		}
+		seen[cur] = struct{}{}
+
+		cur = types.Unalias(cur)
+
+		switch t := cur.(type) {
+		case *types.Pointer:
+			cur = t.Elem()
+			continue
+		case *types.Named:
+			obj := t.Obj()
+			if obj != nil && obj.Pkg() != nil {
+				if obj.Pkg().Path() == "sync" && (obj.Name() == "Mutex" || obj.Name() == "RWMutex") {
```

Comment:
Will this work for the gVisor codebase, which uses import path `gvisor.dev/gvisor/pkg/sync`?
https://github.com/search?q=repo%3Agoogle%2Fgvisor+%22%5C%22gvisor.dev%2Fgvisor%2Fpkg%2Fsync%5C%22%22&type=code